File: java/src/main/java/com/example/backend/adapter/driving/TestResponseCodeController.java
package com.example.backend.adapter.driving;

import jakarta.servlet.http.HttpServletResponse;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.*;

/**
 * 测试响应状态码 Controller
 */
@RestController
public class TestResponseCodeController {

    private static final Logger logger = LoggerFactory.getLogger(TestResponseCodeController.class);

    ///**
    // * 测试响应 500
    // */
    //@ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    //@RequestMapping(
    //        value = "/api/test/response/code/500",
    //        method = RequestMethod.GET,
    //        produces = "text/plain;charset=UTF-8"
    //)
    //public String testResponseCode500()
    //{
    //    return "测试500错误";
    //}

    /**
     * 测试响应 500
     */
    @RequestMapping(
            value = "/api/test/response/code/500",
            method = RequestMethod.GET,
            produces = "text/plain;charset=UTF-8"
    )
    public String testResponseCode500(HttpServletResponse response)
    {
        response.setStatus(500);
        return "测试500错误";
    }
}

File: dotnet/src/adapter/driving/TestResponseCodeController.cs
using System.Dynamic;
using BackendExample.Utility;
using Microsoft.AspNetCore.Mvc;

namespace BackendExample.Adapter.Driving
{
    /// <summary>
    /// 测试响应状态码 Controller
    /// </summary>
    public class TestResponseCodeController : ControllerBase
    {
        /// <summary>
        /// 测试响应 500
        /// </summary>
        [HttpGet("/api/test/response/code/500")]
        [Produces("text/plain")]
        public IActionResult testResponseCode500()
        {
            return StatusCode(500, "测试500错误");
        }

    }
}
